packages/api/src/server/auth-action/auth-actions.dto.ts

```diff
@@ -0,0 +1,18 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsNotEmpty, IsString } from 'class-validator';
+
+export class AuthActionDto {
```

Comment:
I don't see any specific usage for this DTO, i.e., it is not dealing with any transformation or parsing, simply copying its content to the AuthAction class. From the other class [AuthActionFactory](https://github.com/spaship/spaship/pull/2104/files#diff-d0000418748765a321be79de5e4f169730bb2d9e771a56fa2167c0f3e8f615c2R7) I can see it is unnecessarily increasing the number of operations and making the code less readable, Is there any specific purpose which this DTO is solving here?

Reply:
Sure we can discuss on this

Reply:
We're following the same Structure in the Entire project, but as AuthActionsFactory and Role are the instal service so here we may ignore the DTO, but as in the Swagger we are only adding the DTO classes as the Request template, so it will add a discrepancy.

Reply:
That's not a good justification; for Role and AuthActionMapping, the DTOs are not doing anything. Other than adding overhead to operation and code, find a way to mitigate this issue. Till then, add a TODO for the same on code.